src/nimble_serial.cc

```diff
@@ -540,48 +498,15 @@ int ExplicitTimeIntegrator(nimble::Parser & parser,
     }
 
 #ifdef NIMBLE_HAVE_UQ
-   for(int nuq=0; nuq<uq_model.num_samples_; nuq++){
-     double * disp_this_sample = offnominal_displacements[nuq];
-     double * velocity_this_sample = offnominal_velocities[nuq];
-     for (int i=0 ; i<num_unknowns ; ++i) {
-       disp_this_sample[i] += delta_time * velocity_this_sample[i];
-     }
-   }
-#endif
-
-#ifdef NIMBLE_HAVE_UQ
-    if(uq_enabled) {
-      // HACK specific to 1D & linear in parameter
-      // Compute displacement sensitivites, i.e. \partial(U)/\partial(\lambda)
-      // Loop over parameters and do one-by-one
-      for(int np=0; np<uq_parameters.GetNumParams(); np++){
-        int left_sample = uq_parameters.GetLeftAdjSampleForParam(np);
-        int right_sample = uq_parameters.GetRightAdjSampleForParam(np);
-
-        int left_off_nom_disp_field_id = macroscale_data.GetFieldId( "off_nom_displacement_"+std::to_string(left_sample) );
-        int right_off_nom_disp_field_id = macroscale_data.GetFieldId("off_nom_displacement_"+std::to_string(right_sample));
-
-        double* left_off_nom_disp = macroscale_data.GetNodeData(left_off_nom_disp_field_id);
-        double* right_off_nom_disp= macroscale_data.GetNodeData(right_off_nom_disp_field_id);
-
-        if(left_sample != right_sample) { // protect against nsamples ==1
-          uq_parameters.ComputeDisplacementSensitivity(np, num_unknowns, displacement, left_off_nom_disp, right_off_nom_disp,
-                                                       displacement_sensitivities[np]);
-        }
-      }
-    }
+   // advance approximate trajectories
+   uq_model.UpdateDisplacement(delta_time);
+   uq_model.Prep();
```

Comment:
Guessing that re-initialization of off nominal internal forces to 0.0 is happening in uq_model.Prep().